MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -3064,7 +3064,7 @@ public void readNews() {
     }
 
     public int getDeploymentDeficit(AtBContract contract) {
-        if (!contract.isActive()) {
+        if (!contract.getStatus().isActive()) {
```

Comment:
I would suggest retaining the isActive() method, but having it call getStatus().isActive(). That way you can reduce the footprint of this PR quite a bit while keeping the spirit.

Reply:
That is purposeful. I'd rather have the code in the enum in question to prevent future problems... This is especially for tracing, as I've seen what happens when coders go over-the-top with putting methods that do this and the sheer amount of effort that it adds to tracing [especially for enums], and thus would prefer to avoid it in our code base.

Reply:
My preference would be Contract::isActive as well. It is more intuitive and benefits from intellisense (etc). Much like Person::isDeployed which wraps a similar check for convenience.

Reply:
A compromise could be `Contract::isActiveOn(LocalDate date)` which would handle a lot of these call sites.

Reply:
I'm actually planning on implementing contact end date nag and removing most uses of isActive in the near future. This originally contained them, but it got too busy for the PR so I reverted it.

Futher, Person::isDeployed isn't an enum check but rather a unit value check. Because it is an enum check is why this is implemented as it is.

As for isActiveOn, that's a nice way to handle it that will simplify some of the logic for my next changes, but it should be handled by getActiveContract if it is simply an active state check that does not currently need to be handled based only on start date.